spirv_msl.cpp

```diff
@@ -3016,6 +3322,121 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		break;
 	}
 
+	case OpInBoundsAccessChain:
+	case OpAccessChain:
+	case OpPtrAccessChain:
+		if (get_execution_model() == ExecutionModelTessellationControl)
+		{
+			// If this is a per-vertex output, remap it to the I/O array buffer.
+			auto *var = maybe_get<SPIRVariable>(ops[2]);
+			BuiltIn bi_type = BuiltIn(get_decoration(ops[2], DecorationBuiltIn));
+			if (var && (var->storage == StorageClassInput || var->storage == StorageClassOutput) &&
+			    !has_decoration(ops[2], DecorationPatch) &&
+			    (!is_builtin_variable(*var) || bi_type == BuiltInPosition || bi_type == BuiltInPointSize ||
+			     bi_type == BuiltInClipDistance || bi_type == BuiltInCullDistance ||
+			     get_variable_data_type(*var).basetype == SPIRType::Struct))
+			{
+				uint32_t length = instruction.length;
+				AccessChainMeta meta;
+				std::vector<uint32_t> indices;
+				uint32_t next_id = ir.increase_bound_by(2);
+
+				indices.reserve(length - 3 + 1);
+				uint32_t type_id = next_id++;
+				SPIRType new_uint_type;
+				new_uint_type.basetype = SPIRType::UInt;
+				new_uint_type.width = 32;
+				set<SPIRType>(type_id, new_uint_type);
+
+				indices.push_back(ops[3]);
+
+				uint32_t const_mbr_id = next_id++;
+				uint32_t index = get_extended_decoration(ops[2], SPIRVCrossDecorationInterfaceMemberIndex);
+				if (var->storage == StorageClassInput ||
+				    has_decoration(get_variable_element_type(*var).self, DecorationBlock))
+				{
+					uint32_t i = 4;
+					if (index == (uint32_t)-1)
```

Comment:
Prefer C++-style arithmetic cast `uint32_t(-1)` instead.

Reply:
I noticed that, but forgot to go back and fix that...